### PR TITLE
Git: Checkout to a tracked local branch instead when trying to checkout to a remote branch

### LINKED ISSUE #60904: Git fails to checkout remote branch with the same name as a local branch
- VSCode Version: 1.29.0-insider (user setup)
- OS Version: Windows 10 10.0.17134

**more details - entire help entry**
Version: 1.29.0-insider (user setup)
Commit: 931c22033db3cc8d5dff265889307ed4076ac2d3
Date: 2018-10-12T05:17:15.739Z
Electron: 2.0.11
Chrome: 61.0.3163.100
Node.js: 8.9.3
V8: 6.1.534.41
Architecture: x64

Steps to Reproduce:

1. open *vscode* with a git controlled folder.
2. from any branch select the scm icon in status bar to checkout a different branch and select origin/master in the command pallette popup while on say develop

Does this issue occur when all extensions are disabled?: Yes/No
Yes

*this is the error I recieve*
image

**Here is the git log**
```
> git checkout -q --track origin/master
fatal: A branch named 'master' already exists.
> git fetch
> git status -z -u
> git symbolic-ref --short HEAD
> git rev-parse develop
> git rev-parse --symbolic-full-name develop@{u}
> git rev-list --left-right develop...refs/remotes/origin/develop
> git for-each-ref --format %(refname) %(objectname) --sort -committerdate
> git remote --verbose
```  

I normally do not use the insiders build but I installed it to test this out.

There may be a issue with my git for windows configuration and not vscode.
I recently updated my git for windows install and this problem started after the update.

If I simply execute `git checkout master` then the checkout works as expected it seems the problem only happens when using vscode scm features.

### PR DESCRIPTION
**Pull request by laphets**
Fix #60904 
## Description
Problem raised when trying to checkout a remote branch, while there existing a local branch having the same name. 
## Solution
Basically, we can check whether the remote branch name matches with any local branch. However, I think it's confusing sometimes when the local branch is tracking to an upstream branch with another name. Therefore, I think it's better to judge by the **tracking ship** instead of simply **branch name**.
By using
```
git for-each-ref --format='%(refname:short) -> %(upstream:short)' refs/heads
```
we can get all the tracking relations from local branch to the upstream branch, just as what is shown in the capture below.
image

Therefore, I add a `GetTracking` method in `extensions/git/src/git.ts` to parse for the tracking relationship and return.  And when a user is trying, for instance, checkout to `origin/master`, vscode will find the tracking relation from `master` to `origin/master`, and checkout to `master` instead.

### CODE REVIEW

File: extensions/git/src/api/git.d.ts [hunk 1]
```diff
@@ -26,6 +26,11 @@ export interface Ref {
 	readonly remote?: string;
 }
 
+export interface Trackingship {
+	readonly local: string;
+	readonly upstarem: string;
```
Comment (WenInCode) [inline extensions/git/src/api/git.d.ts]: 👋 I think you have a typo here
```suggestion
	readonly upstream: string;
```

**Comment by laphets** [comment]
@joaomoreno Could you help me check this, thx..

**Comment by Astrantia** [comment]
@joaomoreno this is vital

**Comment by joaomoreno** [comment]
Thanks! :fireworks:

**Pull request merged**